src/components/Game.jsx

```diff
@@ -1,34 +1,76 @@
 import React, { Component } from 'react'
-import {shuffleBoard} from './Common/shuffleBoard'
+import {shuffleBoard, copyBoard} from './Common/shuffleBoard'
 import Board from './Board'
 import ScoreBox from './ScoreBox'
 import CurrentWord from './CurrentWord'
 import './Game.css'
 class Game extends Component {
 
-  constructor() {
-    super()
-    this.board = shuffleBoard()
-    this.handleClick = this.handleClick.bind(this)
-    this.handleSubmit = this.handleSubmit.bind(this)
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.initBoard = shuffleBoard();
+    this.surroundingCells = [
+      {r:-1, c:-1},
+      {r:-1, c:0},
+      {r:-1, c:1},
+      {r:0, c:-1},
+      {r:0, c:1},
+      {r:1, c:-1},
+      {r:1, c:0},
+      {r:1, c:1}];
     this.state = {
+      board : this.initBoard,
       currentWord: '',
+      currentWordPosition: [],
       wordList: {},
       totalScore : 0
     }
   }
+  // 1. click on the tile
+  // 2. update tile selected to true.
+  // 2.1 Can select and unselect the tile
+  // 2.2 Can only unselect the last tile
+  // 2.3 Update currentWord as we select and unselect
+  // 2.4. TODO: should be only able to select the surrounding cells
+  // 2.5 Make a copy of board, word, currentWordPositions, etc
+  // 2.6 Mutate the state
+  // 3. render the board with updated tile so it renders as active
+  handleClick(rowId, columnId) {
+    const newBoard = copyBoard(this.state.board);
```

Comment:
There's a lot of copying here.  Can this be avoided by checking if the row/column is invalid first?